curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -535,94 +538,130 @@ bool FileCacheManager::ReadKVRequestFromS3(const std::string &name,
     if (*ret < 0) {
         LOG(ERROR) << "object " << name << " read from s3 fail, ret = " << *ret;
         return false;
-    } else if (s3ClientAdaptor_->s3Metric_.get()) {
+    }
+
+    if (s3ClientAdaptor_->s3Metric_) {
         s3ClientAdaptor_->CollectMetrics(
             &s3ClientAdaptor_->s3Metric_->adaptorReadS3, length, start);
     }
 
     return true;
 }
 
-int FileCacheManager::ReadKVRequest(
-    const std::vector<S3ReadRequest> &kvRequests, char *dataBuf,
-    uint64_t fileLen) {
-
-    for (auto req = kvRequests.begin(); req != kvRequests.end(); req++) {
-        VLOG(6) << "read from kv request " << req->DebugString();
-
-        uint64_t chunkIndex = 0, chunkPos = 0, blockIndex = 0, blockPos = 0;
-        uint64_t chunkSize = s3ClientAdaptor_->GetChunkSize();
-        uint64_t blockSize = s3ClientAdaptor_->GetBlockSize();
-        GetBlockLoc(req->offset, &chunkIndex, &chunkPos, &blockIndex,
-                    &blockPos);
+void FileCacheManager::ProcessKVRequest(const S3ReadRequest &req, char *dataBuf,
+                                        uint64_t fileLen,
+                                        std::once_flag &cancelFlag,
+                                        std::atomic<bool> &isCanceled,
+                                        std::atomic<int> &ret) {
```

Comment:
forget set ret？

Reply:
It will be set when read from s3 failed.